File: hackerrank/problem solving/algorithms/1 - warmup/9 - birthday cake candles/with morfo library/birthday_cake_candles.rb
# https://www.hackerrank.com/challenges/birthday-cake-candles/problem?isFullScreen=true

require 'morfo'

def main
  Morfo.skip_input_ln
  candles = Morfo.readln(&:to_i)
  puts birthday_cake_candles(candles)
end

# n: length of candles
# T: O(2n) => O(n)
# S: O(1) extra space
def birthday_cake_candles(candles) = candles.count(candles.max)

main if __FILE__ == $0

File: hackerrank/problem solving/algorithms/1 - warmup/9 - birthday cake candles/with morfo library/birthday_cake_candles.rs
// https://www.hackerrank.com/challenges/birthday-cake-candles/problem?isFullScreen=true

use morfo::{read_ln, skip_input_ln};

fn main() {
    skip_input_ln();
    let candles: Vec<i32> = read_ln();
    println!("{}", birthday_cake_candles(&candles));
}

// n: length of candles
// T: O(2n) => O(n)
// S: O(1) extra space
fn birthday_cake_candles(candles: &[i32]) -> usize {
    let max = candles.iter().max().unwrap();
    candles.iter().filter(|&x| x == max).count()
}
